### PR TITLE
ts-667 add benchmark for newTransaction(), transaction.send() and checkpoint()

### PR DESCRIPTION
**Pull request by tomskikh**
#667

### CODE REVIEW

File: src/test/scala/com/bwsw/tstreams/benchmark/ProducerBenchmark.scala [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+    val newTransaction = new ExecutionTimeMeasurement
+    val send = createIf(dataSize > 0)
+    val checkpoint = createIf(withCheckpoint)
+
+    for (i <- 1 to iterations) {
+      val transaction = newTransaction(() => producer.newTransaction(NewProducerTransactionPolicy.ErrorIfOpened))
+      send.foreach(_.apply(() => transaction.send(data)))
+      checkpoint.foreach(_.apply(() => producer.checkpoint()))
+
+      if (progressReportRate > 0 && i % progressReportRate == 0)
```
Comment (placccebo) [inline src/test/scala/com/bwsw/tstreams/benchmark/ProducerBenchmark.scala]: let's create "ProgressBar" class with show() method

File: src/test/scala/com/bwsw/tstreams/benchmark/ProducerBenchmarkRunner.scala [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    partitions = config.partitions())
+
+  println(config.dataSize())
+
+  private val result = benchmark.run(config.iterations(), dataSize = config.dataSize())
+  benchmark.close()
+  println("done")
+  println(result)
+
+  System.exit(0)
```
Comment (placccebo) [inline src/test/scala/com/bwsw/tstreams/benchmark/ProducerBenchmarkRunner.scala]: why doesn't the system finish the work as usual (without exit())?

File: src/test/scala/com/bwsw/tstreams/benchmark/ProducerBenchmark.scala [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+import com.bwsw.tstreams.agents.producer.NewProducerTransactionPolicy
+import com.bwsw.tstreams.env.{ConfigurationOptions, TStreamsFactory}
+
+/**
+  * Provides method to measure time of [[com.bwsw.tstreams.agents.producer.Producer]]'s methods
+  *
+  * @param address    ZooKeeper address
+  * @param token      authentication token
+  * @param prefix     path to master node in ZooKeeper
+  * @param stream     stream name. Stream will be created if it not exists.
```
[the remark this answers is not in the document]
Comment (tomskikh) [reply src/test/scala/com/bwsw/tstreams/benchmark/ProducerBenchmark.scala]: fix

File: src/test/scala/com/bwsw/tstreams/benchmark/ProducerBenchmark.scala [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+  * @param partitions amount of partitions on stream
```
[the remark this answers is not in the document]
Comment (tomskikh) [reply src/test/scala/com/bwsw/tstreams/benchmark/ProducerBenchmark.scala]: fixed

File: src/test/scala/com/bwsw/tstreams/benchmark/ProducerBenchmark.scala [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+  if (!storageClient.checkStreamExists(stream))
+    storageClient.createStream(stream, partitions, streamTtl, "benchmark")
+
+
+  /**
+    * Measures time of [[com.bwsw.tstreams.agents.producer.Producer]]'s methods
+    *
+    * @param iterations         amount of measurements
+    * @param dataSize           size of data sent in each transaction
+    * @param withCheckpoint     if true, for each transaction checkpoint will be performed
```
[the remark this answers is not in the document]
Comment (tomskikh) [reply src/test/scala/com/bwsw/tstreams/benchmark/ProducerBenchmark.scala]: fixed

File: src/test/scala/com/bwsw/tstreams/benchmark/ProducerBenchmark.scala [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+    * @param progressReportRate progress will be printed in console after each progressReportRate iterations
```
[the remark this answers is not in the document]
Comment (tomskikh) [reply src/test/scala/com/bwsw/tstreams/benchmark/ProducerBenchmark.scala]: fixed

File: src/test/scala/com/bwsw/tstreams/benchmark/ProducerBenchmark.scala (hunk 1 above)
Comment (tomskikh) [reply src/test/scala/com/bwsw/tstreams/benchmark/ProducerBenchmark.scala]: fixed

File: src/test/scala/com/bwsw/tstreams/benchmark/ProducerBenchmarkRunner.scala (hunk 2 above)
Comment (tomskikh) [reply src/test/scala/com/bwsw/tstreams/benchmark/ProducerBenchmarkRunner.scala]: some threads still working after finish

Comment (placccebo) [reply src/test/scala/com/bwsw/tstreams/benchmark/ProducerBenchmarkRunner.scala]: it's a pity.

**Pull request merged**